konduit-serving-vertx/src/main/java/ai/konduit/serving/vertx/settings/DirectoryFetcher.java

```diff
@@ -48,6 +48,34 @@ public static File getWorkingDir() {
         );
     }
 
+    /**
+     * Creates the build directory if missing and fetches it.
+     * @return konduit-serving build directory.
+     */
+    public static File getBuildDir() {
+        return createAndValidateDirectory(
+                fetchValueBasedOnPriority(
+                        System.getenv(EnvironmentConstants.BUILD_DIR),
+                        System.getProperty(PropertiesConstants.BUILD_DIR),
+                        getDefaultBuildDir()
+                )
+        );
+    }
+
+    /**
+     * Creates the profiles directory if missing and fetches it.
+     * @return konduit-serving profiles directory.
+     */
+    public static File getProfilesDir() {
+        return createAndValidateDirectory(
+                fetchValueBasedOnPriority(
+                        System.getenv(EnvironmentConstants.PROFILES_DIR),
+                        System.getProperty(PropertiesConstants.PROFILES_DIR),
+                        getDefaultProfilesDir()
```

Comment:
For build dir - we should probably put it in a subdirectory (not just unique filenames as you've done in ServeBuildCommand) to avoid the risk of gradle's files causing problems due to the same location for different launches (or between concurrent launches).